### PR TITLE
Remove unwanted code from project templates

### PR DESCRIPTION
**Pull request by wjk**
Inspired by https://github.com/microsoft/ProjectReunion/discussions/574.

### CODE REVIEW

**Comment by dotMorten** [comment]
Love it, except I don't get the point of the button. I'd prefer that to be removed too. It's throw away code. Don't add throw-away to a starter template.

**Comment by wjk** [comment]
The check failures regarding Helix appear to be because I’m a contributor outside of Microsoft, and as such do not have permission to kick off Helix runs (whatever they are), but the build tried anyway. We should take a look at this sometime.

**Comment by mrlacey** [comment]
I see the arguments for making the changes, but I'd want to know the reasons behind adding that code originally before debating whether to remove it.

**Comment by riverar** [comment]
Agreed, feels like the changes here are purely subjective. At the very least, this should go through the proper workflow for public discourse. We've skipped that step here.

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/MainWindow.h [hunk 1]
```diff
@@ -15,8 +15,6 @@ namespace winrt::$safeprojectname$::implementation
 
         int32_t MyProperty();
         void MyProperty(int32_t value);
```
Comment (Jaiganeshkumaran) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/MainWindow.h:17]: I think these properties should be removed. They are useless and everytime you create a new window, you're going to just get rid of it so no use.

File: dev/VSIX/ProjectTemplates/UWP/CppWinRT/BlankApp/MainPage.h [hunk 2]
```diff
@@ -10,8 +10,6 @@ namespace winrt::$safeprojectname$::implementation
 
         int32_t MyProperty();
         void MyProperty(int32_t value);
```
Comment (Jaiganeshkumaran) [inline dev/VSIX/ProjectTemplates/UWP/CppWinRT/BlankApp/MainPage.h:12]: These properties should also be removed

**Review by Jaiganeshkumaran — CHANGES_REQUESTED** [review]

**Comment by Jaiganeshkumaran** [comment]
The C++ version of item templates and project templates seem to nest the source file as top-level if it doesn't have a XAML file rather than the IDL or header file. This is inconsistent with current templates included in the C++/WinRT VSIX and others built-in to Visual Studio.

**Comment by BenJKuhn** [comment]
@MikeHillberg, can you review this PR or find someone to help drive it to closure?

**Comment by dotMorten** [comment]
Any update on this? The current template is driving me bonkers :-D

**Comment by aepot** [comment]
Current templates confused me. Why I must clean-up  xaml and code-behind before everything?

I'm using MVVM+IoC patterns which means no code-behind is needed, except ViewModel property. StackPanel instead of Grid looks suspicious too. First thought was "no Grid support?".

As SDK user I like the changes and suggest to merge the PR.

**Comment by Jaiganeshkumaran** [comment]
Templates now seem to include Microsoft copyright and MIT License notice? Any reason why this was done. It makes sense for generated files to be copyrighted, but templates shouldn't be (obviously you'll add your own code after creating an instance of the template and that's not copyrighted by Microsoft!).

**Comment by bpulliam** [comment]
Closing PRs that appear to have been abandoned.

**Comment by riverar** [comment]
Bob's closure of this PR is part of an slow moving effort to clean up the repository. This PR skipped the proper workflow (at the time) for public discourse/changes, has minor and subjective changes in it, and has had no real engagement since 2021. Seems like this fits the bill for closure to me.

If the author feels strongly about this change, I recommend opening an issue to discuss and get sign off before making template changes.

**Comment by bpulliam** [comment]
@riverar is right, but I also should add more context.
1) This original intent of this PR is to change the templates in a pretty fundamental way - empty content vs single button content. This should be resolved in a discussion first, and then we can look at the PR.  I do see a significant number of bug reports that use the template to attach some functionality to the button, and those would have to create content instead of having other remove it.  These templates were based on C++ winrt templates so there is some history there.  Not saying we can't change, but we should be deliberate about it.
2) This also had become a catch-all for template issues.  We have addressed some of those in #3324 and #3493 already.
3) You are absolutely correct that this conversation should have happened a year ago.

I am happy to have the conversation about this, or any other issue we resolve or close, as the goal is to bring additional clarity on those issues.

**Comment by JaiganeshKumaran** [comment]
> I do see a significant number of bug reports that use the template to attach some functionality to the button, and those would have to create content instead of having other remove it. These templates were based on C++ winrt templates so there is some history there. Not saying we can't change, but we should be deliberate about it.

Apart from bug reports, you are mostly going to delete them anyway. It's only wasting time for you. I did open a pull request to remove them from the C++/WinRT (UWP) templates but it was not merged due to not having enough people to main the VSIX. Windows App SDK, however, is well maintained and this change should be merged. Maybe as an example, the stack panel and button can be included in MainPage.xaml created with the project template, but not in item templates for page or user control.

**Comment by riverar** [comment]
@JaiganeshKumaran "Well maintained" [spits out coffee].

**Comment by JaiganeshKumaran** [comment]
@riverar  Sure is more well-maintained than C++/WinRT, which is done for now.

**Comment by dotMorten** [comment]
> These templates were based on C++ winrt templates so there is some history there.

History? None of the WPF, UWP or WinForms templates ship with anything in them. They are blank and ready to throw content on. It's been like this for many many years.

But yes we should have an issue to discuss this in I guess. I was merely pointing out the messaging of having a PR discussed, then [assigned to a Microsoft team member](https://github.com/microsoft/WindowsAppSDK/pull/622#issuecomment-871112147) to follow it through, only to be closed years later after no one actually following through on it, and no explanation why it was closed besides "stale". It's a really bad messaging to send to community members who spent time trying to improve the SDK, and there's nothing the community could have done here to prevented it from going stale.

**Pull request closed without merging**